### PR TITLE
Verify translator used for import test

### PR DESCRIPTION
**Pull request by retorquere**
In light of the earlier discussion on what the tests represent: this change verifies the use of the translator that the tests were specified in.

### CODE REVIEW

**Comment by retorquere** [comment]
Does this mean I have to rebase to `upstream/master` :cry:?

**Comment by dstillman** [comment]
You can, but there are no conflicts, so it's not necessary. The advantage of rebasing is that it ensures that you've tested on the current version, but if there's no relevant functionality difference, it doesn't matter.

**Comment by dstillman** [comment]
Now there's a conflict, because you changed the same line that was changed on the master branch.

If this depends on the change in master, you need to rebase. If it doesn't, you don't, but then you can't change the same line.

File: test/import_test.js [hunk 1]
```diff
@@ -23,6 +25,24 @@ function addTest(name, input, expected) {
...
 
+function parseJSON(name, json, source) {
+	try {
+		return JSON.parse(json);
+	} catch (err) {
+		it(`should import ${name}`, function () {
+			const m = err.message.match(/(.*) at position ([0-9]+)$/);
+			if (m && typeof cases === 'string') {
+				const pos = parseInt(m[2]);
+				const line = source.substring(0, source.indefOf(json) + pos).split('\n').length;
```
Comment (dstillman) [inline test/import_test.js]: That's definitely not right.

File: src/importEndpoint.js [hunk 2]
```diff
@@ -46,15 +41,14 @@ class ImportEndpoint {
 		var items = await translate.translate({ libraryID: 1 });
 		
 		ctx.set('Content-Type', 'application/json');
+		ctx.set('translator-id', translators[0].translatorID);
```
Comment (dstillman) [inline src/importEndpoint.js]: Let's go with `Zotero-Translator-ID`.

File: test/import_test.js [hunk 3]
```diff
@@ -31,35 +44,21 @@ describe.skip("/import", function () {
...
-					const line = code.substring(0, start + pos).split('\n').length;
-					const column = pos - cases.lastIndexOf('\n', pos);
-					throw new Error(`${m[1]} around line ${line}, column ${column}`);
-				}
-				throw err;
-			});
+		let caseNo = 0;
+		for (const test of cases) {
+			caseNo += 1;
+			if (test.type === 'import') addTest(`${name} # ${caseNo}`, header.translatorID, test.input, test.items);
```
Comment (dstillman) [inline test/import_test.js]: New line and braces (or `if (test.type != 'import') continue;`)

**Review by dstillman — CHANGES_REQUESTED** [review]

File: test/import_test.js (hunk 3 above)
Comment (retorquere) [reply test/import_test.js]: Done, but the .eslintrc proposed for the translators allows this form.

Comment (dstillman) [reply test/import_test.js]: It does, but we try to keep to a line length of 100, and we mostly only use the single-line format for lines like `if (foo) continue;` where readability isn't a concern.

**Comment by dstillman** [comment]
Thanks!

File: test/import_test.js (hunk 3 above)
Comment (retorquere) [reply test/import_test.js]: Is it likely https://github.com/zotero/translators/pull/1895 is going to be merged soon? I want to get started on fixing the import tests but some of these will involve code changes, and I'd prefer to have the linter at hand for that.

**Pull request merged**